### PR TITLE
[FIX] web, website: create many2one widget with auto save

### PR DESCRIPTION
**Pull request by Guillaume-gdi**
This commit introduces a new widget for many2one fields that will automatically save the record when the user changes the value of the field. This was needed for the website settings, where the user could change the website without saving the record.

Steps to reproduce the issue solved by the new widget:
- Have multiple websites
- Go to the website Settings
- Change the name of the "Website 1"
- Change the Website you are editing
- Save

=> The new name of the website 1 has not changed.

task-3265100

### CODE REVIEW

**Comment by Guillaume-gdi** [comment]
Note that [this commit](https://github.com/odoo/odoo/commit/50421af796e9354cbfc6691cba0bf0dc5c989540) fixed the fact that the website selector dirtied the record

**Review by qsm-odoo — COMMENTED** [review]
@rdeodoo I let you handle this one as it seems linked to long discussions with GDI and the framework team. Maybe you can answer this @Guillaume-gdi:
- Why don't we fix the other apps using the same behavior (point of sale apparently? ...?)
- Did we say that we don't care about the "Discard"? Because this fixes the "Save" with more code but breaks the possibility to use "Discard".

**Comment by rdeodoo** [comment]
@Guillaume-gdi I don't remember exactly what was the conclusion here, there was a very long discussion on discord, around and between https://discord.com/channels/678381219515465750/687338039717920792/1119278336485621911 and https://discord.com/channels/678381219515465750/687338039717920792/1118919199788638230 which ends up on a quick call with the framework-js team.

I think the conclusion was that something was missing at the framework level and that this type of m2o field should indeed receive a fix, including the one in website but also in POS and somewhere else I think.

Did we decide to auto save? Wasn't it rather a warning to prevent you switching record with pending change?

My bad on this one, I should have written a summary somewhere.

Could you provide the info you have about what was decided UX wise?

**Comment by rdeodoo** [comment]
> Did we decide to auto save? Wasn't it rather a warning to prevent you switching record with pending change?
Could you provide the info you have about what was decided UX wise?

Once things settle down and we pick this one up again, let's do the warning / prevention of record switching rather than the auto save IMO.

And let's do that for POS too (or anything using this type of m2o switcher), it's a bug for them too :+1: 
As the framework did: https://github.com/odoo/odoo/pull/125550/ global fix

**Comment by Guillaume-gdi** [comment]
Sorry for my late reply... I thought I'd answered, but I didn't. :grimacing: 

At the time, we opted for the auto save solution. I think it was a good decision. It's getting closer to the new fashion of auto save everywhere in Odoo. I even had a discussion with Gerry to make sure it wasn't a problem. But if you really want the other solution, I can do it.

**Comment by rdeodoo** [comment]
If you prefer the auto save, I am fine, I don't like that but as you say it's in par with the Odoo way.

AFAIC:
1. Make it so it's a global fix, even for POS & co, see my previous message, keep fixing as the framework did: globally, not only website
2. Then reach out to the framework team, they should be the one reviewing and merging this one. (be sure to mention / explain the previous fix they make and that this one is the followup)

**Comment by Guillaume-gdi** [comment]
@rdeodoo 
> Make it so it's a global fix, even for POS & co, see my previous message, keep fixing as the framework did: globally, not only website

What do you mean ? You want me to make that every many 2 one auto-save the record or only when it's problematic (website and POS setting only) ?

> Then reach out to the framework team, they should be the one reviewing and merging this one. (be sure to mention / explain the previous fix they make and that this one is the followup)

GED already had a look, but yes when it's ok for you, I will ask them for review + merge :+1:

**Comment by rdeodoo** [comment]
What I meant is that the first fix of the framework (https://github.com/odoo/odoo/commit/50421af796e9354cbfc6691cba0bf0dc5c989540) did that in a generic way which benefits to all "settings headers" (that's how they call it), regardless if it's the website one, the pos one or anything.

Here, you are adding a new type of field/behavior (m2o auto save) but only website benefits from it.
POS (and there is a third one I believe) should also receive the fix.

And it should IMO not require any code in those modules, it should work out of the box, especially since the framework manages to identify such fields and apply a fix generically (setting `doNotSetDirty` on `isHeaderField` fields which are automatically identified).

So, tl;dr:
1. Fix other "settings headers" too
2. Ideally make the fix generic, without any need of those settings headers to be marked or flagged with a class or option or whatever

**Comment by Guillaume-gdi** [comment]
Finally, the framework JS team handled this task with a dialog (save/stay here/discard). See [their PR](https://github.com/odoo/odoo/pull/144005)
